var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -732,10 +733,48 @@ def build(self, spec, prefix):
         with spack.util.environment.set_env(PWD=self.build_directory):
             super(Cp2k, self).build(spec, prefix)
 
+            with working_dir(self.build_directory):
+                make("libcp2k", *self.build_targets)
+
     def install(self, spec, prefix):
         exe_dir = join_path("exe", self.makefile_architecture)
+        lib_dir = join_path("lib", self.makefile_architecture, self.makefile_version)
+
         install_tree(exe_dir, self.prefix.bin)
         install_tree("data", self.prefix.share.data)
+        install_tree(lib_dir, self.prefix.lib)
+
+        mkdirp(self.prefix.include)
+        install("src/start/libcp2k.h", join_path(self.prefix.include, "libcp2k.h"))
+
+    @run_after("install")
+    def fix_package_config(self):
+        """
+        Default build procedure generates libcp2k.pc with invalid paths,
+        because they are collected from temporary directory.
+
+        Ignoring invalid paths, most library-related switches are correct
+        except for fftw and openblas.
+
+        This procedure is appending two missing switches (tested with GROMACS 2022.2 + CP2K).
+
+        In case such approach causes issues in the future, it might be necessary
+        to generate and override entire libcp2k.pc.
+        """
+        with open(join_path(self.prefix.lib.pkgconfig, "libcp2k.pc"), "r+") as handle:
+            content = handle.read().rstrip()
+
+            if "^fftw" in self.spec:
+                content += " -lfftw3"
+
+            if "^fftw+openmp" in self.spec:
+                content += " -lfftw3_omp"
+
+            content += " -lopenblas"
```

Comment:
I believe this line leads to problems when one wants to use something other than openblas, right?
I try to install `gromacs+cp2k ^netlib-lapack` and get `/usr/bin/ld: cannot find -lopenblas: No such file or directory`

Reply:
this has been addressed

Reply:
Yes, but so far only `~mpi` build works. I am trying to make `+mpi` build as well. Not sure where is the issue.

Reply:
@w8jcik post the error and the full linker line leading up to the error here or on Slack, maybe someone has seen the error and can help?

Reply:
> Yes, but only `gromacs~mpi` builds so far. I am trying to make `gromacs+mpi` build as well. Not sure where is the issue, except that it fails during linking.

I see this too. First of many errors I get when building (linking) gromacs is `undefined reference to pilaenv_`. This seems to be some [scalapack function](https://netlib.org/scalapack/explore-html/d7/d8f/pilaenv_8f_source.html). This makes somehow sense, since scalapack is only used with `+mpi` in cp2k. Also I see that there is no `-lscalapack` flag in the command that trows the errors. Not sure how this should be fixed though.

Reply:
maybe the `.pc` file needs some more fixing?

Reply:
Yes, it is most likely fault of `.pc` file.

This `Dockerfile` builds CP2K in Spack-less 

```
FROM ubuntu:22.04

# https://serverfault.com/a/797318
ARG DEBIAN_FRONTEND=noninteractive

RUN apt-get update -y
RUN apt-get upgrade -y

# CUDA 11.5.1 and GCC 11.2
RUN apt-get install -y nvidia-cuda-toolkit nvidia-cuda-dev
RUN apt-get install -y build-essential cmake git

RUN apt-get install -qq --no-install-recommends \
  autoconf autogen automake autotools-dev bzip2 ca-certificates git less \
  libtool libtool-bin make nano patch pkg-config python3 unzip wget xxd zlib1g-dev

RUN apt-get install -y gcc g++ gfortran
RUN apt-get install -y gcc-10 g++-10 gfortran-10

RUN apt-get install -qq libopenblas-dev libfftw3-dev

RUN useradd -m user
USER user
WORKDIR /home/user
SHELL ["/bin/bash", "-c"]

RUN git clone --recursive -b v2022.1 https://github.com/cp2k/cp2k.git

RUN cd "${HOME}/cp2k/tools/toolchain/" \
    && ./install_cp2k_toolchain.sh --with-libxsmm=install \
        --with-openblas=system --with-fftw=system \
        --enable-cuda --gpu-ver=V100

RUN cp "${HOME}/cp2k/tools/toolchain/install/arch/"* "${HOME}/cp2k/arch/"

RUN source "${HOME}/cp2k/tools/toolchain/install/setup" \
    && cd cp2k \
    && make -j ARCH=local VERSION=ssmp \
    && make -j ARCH=local VERSION=ssmp libcp2k

# RUN source "${HOME}/cp2k/tools/toolchain/install/setup" \
#     && cd cp2k \
#     && make -j ARCH=local VERSION=ssmp test

RUN git clone -b v2022.2 https://github.com/gromacs/gromacs.git gromacs-cp2k-src

RUN source "${HOME}/cp2k/tools/toolchain/install/setup" \
    && DIR="gromacs-cp2k" \
    && cmake \
        -DBUILD_SHARED_LIBS=OFF -DGMXAPI=OFF -DGMX_INSTALL_NBLIB_API=OFF \
        -DGMX_EXTERNAL_BLAS=ON -DGMX_EXTERNAL_LAPACK=ON \
        -DGMX_CP2K=ON -DCP2K_DIR="${HOME}/cp2k/lib/local/ssmp" \
        -DGMX_GPU=cuda \
        -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10 -DCMAKE_FORTRAN_COMPILER=gfortran-10 \
        -DCMAKE_INSTALL_PREFIX=`pwd`/${DIR} -S "${DIR}-src" -B "${DIR}-build" \
    && cmake --build "${DIR}-build" --target install -j $(nproc)

RUN source gromacs-cp2k/bin/GMXRC.bash \
    && gmx --version
```

My idea is to change it to build with MPI and see how the `.pc` file looks.

Other, maybe simpler option is to analyse the script that generates `.pc`.

I will try in some time, otherwise you might find this `Dockerfile` useful.